test/e2e/helpers.js

```diff
@@ -100,29 +99,79 @@ async function withFixtures(options, testSuite) {
   let webDriver;
   let driver;
   let failed = false;
+
+  // We allow to pass both a string, like 'anvil' and array ,like ['anvil']
+  const localNodes = Array.isArray(localNode) ? localNode : [localNode];
+  const localNodesOptions = Array.isArray(localNodeOptions)
+    ? localNodeOptions
+    : [localNodeOptions];
+  let ganacheServer;
+  let anvilServer;
+  const servers = [];
+
   try {
-    if (!disableGanache) {
-      await ganacheServer.start(localNodeOptions);
+    // Start servers based on the localNodes array
+    for (let i = 0; i < localNodes.length; i++) {
+      const node = localNodes[i];
+      const opts = localNodesOptions[i] || {};
+
+      switch (node) {
+        case 'anvil':
+          anvilServer = new Anvil();
+          await anvilServer.start(opts);
+          servers.push(anvilServer);
+          break;
+
+        case 'ganache':
+          ganacheServer = new Ganache();
+          await ganacheServer.start(opts);
+          servers.push(ganacheServer);
+          break;
+
+        case 'none':
+          break;
+
+        default:
+          throw new Error(
+            `Unsupported localNode: '${node}'. Cannot start the server.`,
+          );
+      }
     }
-    let contractRegistry;
 
-    if (smartContract && !disableGanache) {
-      const ganacheSeeder = new GanacheSeeder(ganacheServer.getProvider());
+    let contractRegistry;
+    let seeder;
+
+    // We default the smart contract seeder to the first node client
+    // If there's a need to deploy multiple smart contracts in multiple clients
+    // this assumption is no longer correct and the below code needs to be modified accordingly
+    if (smartContract) {
+      switch (localNodes[0]) {
+        case 'anvil':
+          seeder = new AnvilSeeder(anvilServer.getProvider());
+          break;
+
+        case 'ganache':
+          seeder = new GanacheSeeder(ganacheServer.getProvider());
+          break;
+
+        default:
+          throw new Error(
+            `Unsupported localNode: '${localNodes[0]}'. Cannot deploy smart contracts.`,
+          );
+      }
       const contracts =
         smartContract instanceof Array ? smartContract : [smartContract];
       await Promise.all(
-        contracts.map((contract) =>
-          ganacheSeeder.deploySmartContract(contract),
-        ),
+        contracts.map((contract) => seeder.deploySmartContract(contract)),
       );
-      contractRegistry = ganacheSeeder.getContractRegistry();
+      contractRegistry = seeder.getContractRegistry();
     }
 
     await fixtureServer.start();
     fixtureServer.loadJsonState(fixtures, contractRegistry);
 
-    if (localNodeOptions?.concurrent) {
-      localNodeOptions.concurrent.forEach(async (ganacheSettings) => {
+    if (localNodesOptions[0]?.concurrent) {
+      localNodesOptions[0].concurrent.forEach(async (ganacheSettings) => {
```

Comment:
:information_source:  this will be removed in the future, once all ganache specs are migrated. Now it's left so the PR has smaller scope (see future work list, in PR description)